src/fu-engine.c

```diff
@@ -5953,20 +5960,37 @@ fu_engine_ensure_security_attrs(FuEngine *self)
 
 	/* distil into one simple string */
 	g_free(self->host_security_id);
-	self->host_security_id = fu_engine_attrs_calculate_hsi_for_chassis(self);
+	self->host_security_id = fu_engine_attrs_calculate_hsi_for_chassis(self, &hsi_number);
+
+	if (fu_history_get_last_hsi_details(self->history,
+					    &previous_hsi,
+					    &last_json_attr,
+					    &error)) {
+		if (previous_hsi != G_MAXUINT) {
+			if (fu_security_attrs_compare_hsi_score(previous_hsi, hsi_number))
+				diff_result =
```

Comment:
you can move the g_autofree gchar *diff_result = NULL; into this scope

Reply:
Also, we're not print the text or anything?

Reply:
diff_result will be referred at line 5991 (fu_history_add_security_attribute()) so I should let it stay in current scope.
Yes, you pointed this out. I'll print some text to notice user HSI number was changed here.